src/entries/extractor-factory.ts

```diff
@@ -0,0 +1,8 @@
+import {WebhookPayload} from '@actions/github/lib/interfaces'
+import {DependabotExtractor} from './dependabot-extractor'
+import {EntryExtractor} from './entry-extractor'
+
+// Determines which extractor to use based on the PR
```

Comment:
@subjunk - the idea here is you would add some logic to choose which extractor to use based on the PR here and then return your extractor. Just needs to conform to the `EntryExtractor` interface